### ISSUE TITLE
Author merge failures display incorrect success message

### ISSUE THREAD

**Issue by seabelis** [internetarchive/openlibrary#8042]
After merging an author profile that fails, a success message is displayed.
### Steps to Reproduce

1. Do ...try to merge an author that is not mergable.

* Actual: If the author is not merged a success message is displayed.
* Expected: If the author is not merged a failure message should be displayed.

### Details

- **Logged in (Y/N)?**
- **Browser type/version?**
- **Operating system?**
- **Environment (prod/dev/local)?** prod

**Comment by tfmorris** [comment]
I've noticed this too. Super annoying.

**Comment by cdrini** [comment]
Go for it @benbdeitch ! I will note a distinction here:

This issue is about the fact that _when_ the endpoint errors, the UI incorrectly displays success. The scope of this issue is **not** to figure out why it is erroring, just to make the UI correctly display the state of the request :+1:

**Comment by cdrini** [comment]
I added a link to the chunk of the file that is likely relevant :+1:

**Comment by tfmorris** [comment]
> The scope of this issue is not to figure out why it is erroring,

@cdrini Good point, but is there a separate issue to address the root cause of the failing author merges? Presumably it's something data dependent.

**Comment by cdrini** [comment]
I believe this is tracking the root cause: #5594

**Comment by benbdeitch** [comment]
Wonderful, thanks for assigning it to me. I'll get to work on it shortly.
